### ISSUE TITLE
Feature request: mouse support for selecting dates

### ISSUE THREAD

**Issue by bovine3dom** [in0rdr/diary#60]
It looks like to could be slightly tricky since we need to convert pixel co-ordinates to dates.

https://tldp.org/HOWTO/NCURSES-Programming-HOWTO/mouse.html

I mention this because `diary` works fine on Android under Termux, but navigating the calendar isn't very nice with the on-screen keyboard. It's probably nicer to be able to tap a date to select it (and Termux fakes mouse events from the touch screen, I think).

While we're there, we could add mouse-wheel support for scrolling up and down if the user has ncurses-6 - https://stackoverflow.com/questions/42356352/mouse-scroll-up-ncurses-c#answer-42357034. It should be as simple as binding mouse 4/5 to what J/K currently do.

Somewhat related to #49.

**Comment by in0rdr** [comment]
Hi @bovine3dom can you review this first draft on your device? Thanks.

Unfortunately I could not get the mouse 4/5 events working (they were not firing on my Ubuntu 20.04, probably, ncurses was not compiled with the feature flag `--enable-ext-mouse`? https://invisible-island.net/ncurses/ncurses-mapsyms.html#ext_mouse).

```
~/Downloads/diary$ ldd diary | grep ncur
	libncursesw.so.6 => /lib/x86_64-linux-gnu/libncursesw.so.6 (0x00007f4cc84dc000)
```

Let's see what can be done...

**Comment by in0rdr** [comment]
The pixel co-ordination to date conversion is not perfect :) I used `BUTTON1_CLICKED` (short click) instead of `BUTTON1_PRESSED` (long press), this might be a bit different on your device..

**Comment by in0rdr** [comment]
implemented in #79
